rts/Sim/Weapons/WeaponDef.cpp

```
@@ -580,21 +580,43 @@ void WeaponDef::LoadSound(
 	GuiSoundSet& soundData
 ) {
 	RECOIL_DETAILED_TRACY_ZONE;
-	switch (hashString(soundKey.c_str())) {
-		case hashString("soundStart"): {
-			CommonDefHandler::AddSoundSetData(soundData, wdTable.GetString(soundKey, ""), wdTable.GetFloat(soundKey + "Volume", 1.0f));
-		} break;
-
-		case hashString("soundHitDry"): {
-			CommonDefHandler::AddSoundSetData(soundData, wdTable.GetString(soundKey, wdTable.GetString("soundHit", "")), wdTable.GetFloat(soundKey + "Volume", wdTable.GetFloat("soundHitVolume", 1.0f)));
-		} break;
-		case hashString("soundHitWet"): {
-			CommonDefHandler::AddSoundSetData(soundData, wdTable.GetString(soundKey, wdTable.GetString("soundHit", "")), wdTable.GetFloat(soundKey + "Volume", wdTable.GetFloat("soundHitVolume", 1.0f)));
-		} break;
-
-		default: {
-		} break;
-	}
+
+    std::string hitFallbackKey = "soundHit";
+
+    uint32_t hash = hashString(soundKey.c_str());
+    bool useHitFallback = hash == hashString("soundHitWet") || hash == hashString("soundHitDry");
```

I somewhat dislike hardcoding these here. Would ideally be something like
```c
LoadSound("soundStart", ...); // no fallback
if (!LoadSound("soundHitDry", ...))
  LoadSound("soundHit", ...);
if (!LoadSound("soundHitWet", ...))
  LoadSound("soundHit", ...);
```

Speaking of which, looks like those lines currently have the same sound set passed into both Wet and Dry and there's these lines
https://github.com/beyond-all-reason/RecoilEngine/blob/b8ef81bcbe1fd5e1bff5934374f2035dd3410467/rts/Game/GameHelper.cpp#L342-L347
So maybe that will need to be touched as well before this works for dry/wet hit sounds.

---

We could make a pre-computed static def for these? The hashStrings should anyways not be done like this on every invocation of the function. If we had a static class or something, then we could reference it project wide and just change it there.

I also wonder, would that then be too specific to BAR for Recoil? Maybe it does need to be a configurable lua file or something...

---

`hashString("foo")` is constexpr so that is probably fine. How would a precomputed static def look?

---

I think something like that would be okay one level higher, at the caller. It sounds cleaner to have a more elementary parse function that just tries to parse what it's given, and separate higher logic that decides what to parse and what to do in case parsing fails (and which could use the new static for that), rather than a monolith that also handles its own fallbacks.

---

I added something now, maybe this is the monolith that you didn't want? :face_holding_back_tears: